service/src/main/java/bio/terra/cbas/initialization/cloneRecovery/CloneRecoveryTransactionService.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.cbas.initialization.cloneRecovery;
+
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.models.RunSet;
+import bio.terra.common.db.WriteTransaction;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CloneRecoveryTransactionService {
+  private final RunDao runDao;
+  private final RunSetDao runSetDao;
+
+  public CloneRecoveryTransactionService(RunSetDao runSetDao, RunDao runDao) {
```

Comment:
This class looks simple enough that unit tests are probably not strictly required, but if you wanted to wire up a couple of mocks in a unit test to prove that “convert to template” was doing the right thing in a couple of situations, and “deleteRunSetAndRuns” was doing the right thing in a couple of situations, I think it would give us a little bit of confidence now and into the future

Reply:
I've added unit tests, but I could not come up with more than one situation to test these methods in -- I think my single test for each method covers everything. But if I'm wrong about that please let me know.